Implementation of CommandFilters, replacing cooldowns and costs

FrankHeijden opened the pull request:
Basically implements #3200, with just a small difference in yaml format: made the `name` required, as a key, just like kits.
Automatic converter is included, and will create the new yaml format from existing nodes.
Formats for `command` will be the same as with the `command-costs`, formats for `pattern` will be the same as in `command-cooldowns`.

Closes #3200, #3847

FrankHeijden committed:
Implementation of CommandFilters, replacing cooldowns and costs

Bobcat00 commented:
How does this affect PR #3744 ?

FrankHeijden commented:
> How does this affect PR #3744 ?

They should go hand in hand with each other, defining a `pattern` filter in this version will behave the same as a regular `command-cooldown` in that PR

FrankHeijden committed:
Add command-filters.yml to .gitignore

FrankHeijden committed:
Change automatic upgrading of command-costs to 'command' instead

Bobcat00 commented:
> They should go hand in hand with each other, defining a `pattern` filter in this version will behave the same as a regular `command-cooldown` in that PR

So this PR needs #3744 ?  Just asking because 3744 hasn't been merged yet.

FrankHeijden commented:
> > They should go hand in hand with each other, defining a `pattern` filter in this version will behave the same as a regular `command-cooldown` in that PR
> 
> So this PR needs #3744 ? Just asking because 3744 hasn't been merged yet.

I mean they are independent features -- #3744 fixes a bug where `commands.yml` defined commands aren't recognised by EssX, this one refactors the cooldowns/costs into a separate yml file as described per #3200 (and by doing that fixing a bug where periods couldn't be used in the regexes, because yml prohibits periods in keys).

FrankHeijden committed:
@FrankHeijden
Make CommandFilter more extendable + fix AsyncTeleport not applying command cooldown

FrankHeijden commented:
So I just came across a thing which prevented AsyncTeleport's from applying command cooldowns - this is due to the fact AsyncTeleport creates a new Trade without the command defined in it. I added a new `cooldownCommand` parameter for the Trade#charge() command, so this method can know what command we should apply cooldowns on.

FrankHeijden committed:
Merge branch '2.x' into extract-command-filters

FrankHeijden committed:
Merge branch '2.x' into extract-command-filters

FrankHeijden committed:
Merge branch '2.x' into extract-command-filters

# Conflicts:
#	Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java

JRoy commented:
Conflicts need to be addressed, please reopen when done so.

FrankHeijden commented:
Should be fixed

FrankHeijden commented:
Cant reopen this one, opening #4224 instead

FrankHeijden committed:
Merge branch '2.x' into extract-command-filters

# Conflicts:
#	Essentials/src/main/java/com/earth2me/essentials/Essentials.java
#	Essentials/src/main/java/com/earth2me/essentials/Settings.java

FrankHeijden committed:
Convert to sponge config

FrankHeijden commented:
What’s the status on this PR? 😅

JRoy commented:
> What’s the status on this PR? 😅

This likely won't make it in time for 2.19 but 100% 2.20/2.19.x

JRoy reviewed (changes requested):
Needs to be updated to configurate (use EssentialsConfiguration over EssentialsConf)

FrankHeijden committed:
Merge branch '2.x' into extract-command-filters

FrankHeijden committed:
Update to configurate

FrankHeijden commented:
> Needs to be updated to configurate (use EssentialsConfiguration over EssentialsConf)

Done

mdcfe commented:
Just a quick heads-up: I have a pending review for this PR, but I likely won't have time to properly review this for a few weeks.

JRoy committed:
Merge branch '2.x' into extract-command-filters

JRoy commented:
Remaining errors (EssentialsConf and logger stuff) need to be fixed. It's almost been 2 years since this PR was made, and I appreciate your persistence 😅. This pull request will be merged for 2.20 which is HOPEFULLY before the new year.

FrankHeijden committed:
Merge branch 'EssentialsX:2.x' into extract-command-filters

FrankHeijden commented:
@JRoy Some build dependencies are timing out because they redirect to jcenter, which closed down. Do they need to be updated or is there a repo missing for these dependencies?

<img width="852" alt="WkyRXerS" src="https://user-images.githubusercontent.com/22407829/198907279-9951db9e-12b4-4e3a-969c-b68d6c3f9b06.png">

JRoy committed:
Merge remote-tracking branch 'upstream/2.x' into extract-command-filters

# Conflicts:
#	Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java

JRoy committed:
Fix build
